### ISSUE TITLE
Setting axis values ignores tickCount

### ISSUE THREAD

**Issue by domoritz** [uwdata/vega-parser#59]
I would like to set the axis values but also want my axis to respect the `tickCount` property. 

This would resolve a problem in Vega-Lite (https://github.com/vega/vega-lite/issues/2942) where we want to have axis ticks that are aligned with the bin boundaries but at the same time, we don't want too many axis ticks. I propose the following algorithm:

* If `values.length` is less than tickCount, use the values.
* Otherwise only use every 2nd tick and repeat until the number of ticks is less than tickCount. 

Luckily, I don't consider this a breaking change as it only changes some undocumented behavior.

**Comment by jheer** [comment]
Makes sense. FWIW that the same algorithm as the default 'parity' form of overlap removal, so that is consistent. This change sounds feasible, but is not part of vega-parser. The modifications should be to the vega-encode repo, most likely by extending the [validTicks](https://github.com/vega/vega-encode/blob/master/src/ticks.js#L40) method to accept a count parameter as well.

**Comment by domoritz** [comment]
Like https://github.com/vega/vega-encode/pull/10?

**Comment by jheer** [comment]
Added as of vega/vega-encode@4aea880703c19eeb2e53456bfae0bcea15f73eab and included in Vega 3.0.3.

**Comment by domoritz** [comment]
It doesn't seem to work with 3.0.4. 

```json
{
  "$schema": "https://vega.github.io/schema/vega/v3.0.json",
  "width": 400,
  "data": [
    {
      "name": "source_0",
      "url": "data/cars.json",
      "format": {"type": "json"},
      "transform": [
        {
          "type": "extent",
          "field": "Acceleration",
          "signal": "bin_maxbins_100_Acceleration_extent"
        },
        {
          "type": "bin",
          "field": "Acceleration",
          "as": [
            "bin_maxbins_100_Acceleration",
            "bin_maxbins_100_Acceleration_end"
          ],
          "signal": "bin_maxbins_100_Acceleration_bins",
          "maxbins": 1000,
          "extent": {"signal": "bin_maxbins_100_Acceleration_extent"}
        }
      ]
    }
  ],
  "scales": [
    {
      "name": "x",
      "type": "linear",
      "domain": {
        "data": "source_0",
        "fields": [
          "bin_maxbins_100_Acceleration",
          "bin_maxbins_100_Acceleration_end"
        ],
        "sort": true
      },
      "range": "width",
      "zero": false
    }
  ],
  "axes": [
    {
      "scale": "x",
      "labelOverlap": true,
      "orient": "bottom",
      "tickCount": 5,
      "values": {
        "signal": "sequence(bin_maxbins_100_Acceleration_bins.start, bin_maxbins_100_Acceleration_bins.stop + bin_maxbins_100_Acceleration_bins.step, bin_maxbins_100_Acceleration_bins.step)"
      }
    }
  ]
}
```

<img width="434" alt="screen shot 2017-09-30 at 11 29 31" src="https://user-images.githubusercontent.com/589034/31048483-a6ffd266-a5d2-11e7-8d56-b24804c10b2a.png">

**Comment by jheer** [comment]
Fix released in vega-encode v2.0.4, will be included in next Vega release (within the next day).
